### PR TITLE
Document classes and member variables unless obvious

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**

### CODE REVIEW

**Comment by owen-jones-diffblue** [comment]
I'd be happy to extend the "should be documented when it is not immediately obvious what they do" language to functions as well, since I think this is what happens in practice, i.e. no one writes documentation for a simple getter or setter.

**Comment by owen-mc-diffblue** [comment]
Note that we are already doing this in some cases. `goto_programt` and `goto_programt::instructiont` are good examples.

**Review by peterschrammel — APPROVED** [review]

File: CODING_STANDARD.md [hunk 1]
```diff
@@ -59,12 +59,13 @@ Formatting is enforced using clang-format. For more information about this, see
   ```
   Note that the `\file` tag must be immediately followed by a newline in order
   for Doxygen to relate the comment to the current file.
-- Each function should be preceded by a Doxygen comment describing that
-  function. The format should match the [LLVM
+- Each class, member variable and function should be preceded by a Doxygen
+  comment describing it when it is not immediately obvious what it does. The
+  format should match the [LLVM
```
Comment (tautschnig) [inline CODING_STANDARD.md:64]: Maybe we also need an explicit note that {class,member,function} names must make clear their purpose?

Comment (owen-mc-diffblue) [reply CODING_STANDARD.md:64]: I've added a commit doing this, but in the _Naming_ section below

**Review by tautschnig — APPROVED** [review]

**Pull request merged**